magellan-library/src/main/java/com/wealthfront/magellan/lifecycle/LifecycleExtensions.kt

```diff
@@ -2,58 +2,58 @@ package com.wealthfront.magellan.lifecycle
 
 import kotlin.reflect.KProperty
 
-public fun <T : LifecycleAware, R> LifecycleOwner.lifecycle(lifecycleAware: T, getter: (T) -> R): Lifecycle<T, R> {
-  return Lifecycle(
+public fun <Child : LifecycleAware, Property> LifecycleOwner.attachFieldToLifecycle(lifecycleAware: Child, getter: (Child) -> Property): LifecycleChild<Child, Property> {
+  return LifecycleChild(
```

Comment:
`Lifecycle` clearly wasn't the best option, but I'm not sure I like `LifecycleChild` any better. Maybe `AttachFieldToLifecycleDelegate`, mirroring the method you call to get one of these?